docs/endpoints/config_reference.rst

```diff
@@ -0,0 +1,297 @@
+Configuration Reference
+***********************
+
+Compute endpoints are configured by a text YAML file named ``config.yaml`` in the
+endpoint directory.  (A reminder that the "endpoint directory" is a subdirectory within
+``~/.globus_compute/``, with the same name as the endpoint.)  This YAML file is a
+convenience interface to the Python configuration classes used internally by Compute
+endpoints.  Anything specified in YAML is conveyed during the endpoint startup to those
+backing classes; consequently, for a complete list of options, please see the
+:ref:`internal class documentation <config-class-doc>` at the bottom of this page.
+Meanwhile, the YAML interface is generally much easier to understand, easier to
+``diff``, and, with less required boilerplate code, easier to maintain.
+
+YAML Configuration
+==================
+
+Compute endpoints currently come in two flavors: one for processing tasks and one for
+*managing* multiple task-processing endpoints.  We discuss the former first, backed by
+|UserEndpointConfig|, as it is typically the first entry into Compute for most people.
+
+User Endpoint Configuration
+---------------------------
+
+The basic configuration has a number of options, but the only required item is
+``engine``.  Globus Compute implements three engines: ``ThreadPoolEngine``,
+``ProcessPoolEngine``, and ``GlobusComputeEngine``.  The first two are Compute endpoint
+wrappers of Python's |ThreadPoolExecutor|_ and |ProcessPoolExecutor|_.  These engines
+are most appropriate for single‑host installations (e.g., a personal workstation).  For
+scheduler‑based clusters, |GlobusComputeEngine|_, as a wrapper over Parsl's
+|HighThroughputExecutor|_, enables access to multiple computation nodes.
+
+The simplest configuration would use the ``ThreadPoolEngine``:
+
+.. code-block:: yaml
+   :caption: ``~/.globus_compute/simple_threadpool/config.yaml``
+
+   engine:
+     type: ThreadPoolEngine
+
+Per Python's default of ``max_workers=None``, this configuration will create as many
+threads as the host has processor cores (up to 32, per Python 3.8+).  Any argument
+to fine-tune the underlying executor's behavior must be placed inside the ``engine``
+stanza.  For example, to limit the worker to 3 threads:
+
+.. code-block:: yaml
+   :caption: ``~/.globus_compute/three_threads/config.yaml``
+
+   engine:
+     type: ThreadPoolEngine
+     max_workers: 3
+
+Similarly, if using the ``ProcessPoolEngine``, one might implement a policy of workers
+only running 100 tasks before workers are respawned:
+
+.. code-block:: yaml
+   :caption: ``~/.globus_compute/four_workers_100_tasks/config.yaml``
+
+   engine:
+     type: ProcessPoolEngine
+     max_tasks_per_child: 100
```

Comment:
I'm not sure if we should cover `ProcessPoolExecutor` options such as these. I'd guess that pointing to [process pool docs](https://docs.python.org/3/library/concurrent.futures.html#concurrent.futures.ProcessPoolExecutor) would be sufficient.

Reply:
*nod; in my mind, this is less about the `ProcessPoolEngine` specifics and more about the basic thrust of what the YAML structure looks like, and contrasting the structure given two different well-known implementations they're specifying (ThreadPoolExecutor and ProcessPoolExecutor).  In my mind, it's much easier to understand something when you know what it's built on top of.

(Note that we _do_ point to those docs, in the opening paragraph of this section.)

Reply:
Cool! I didn't notice that `|ProcessPoolExecutor|_` was a reference.